src/multibase.js

```diff
@@ -0,0 +1,127 @@
+'use strict'
+
+const constants = require('./constants')
+const bs58 = require('bs58')
+
+exports = module.exports = multibase
+exports.encode = encode
+exports.decode = decode
+exports.isEncoded = isEncoded
+
+const errNotSupported = new Error('Unsupported encoding')
+
+// returns a new buffer with the multibase varint+code`
+function multibase (nameOrCode, buf) {
+  if (!buf) {
+    throw new Error('requires an encoded buffer')
+  }
+  const code = getCode(nameOrCode)
+  const codeBuf = new Buffer(code)
+
+  const name = getName(nameOrCode)
+  validEncode(name, buf)
+  return Buffer.concat([codeBuf, buf])
+}
+
+function encode (nameOrCode, buf) {
+  const name = getName(nameOrCode)
+
+  let encode
+
+  switch (name) {
```

Comment:
Is needed to have the `switch` statement if there is just one condition?

```
if (name === 'base58btc') {
  // do stuff
} else {
  throw errNotSupported
}
```

Guessing it's because planned expansion but it would make in that case to solve it with polymorphism rather than conditionals...